### PR TITLE
Cpuinfo support

### PR DESCRIPTION
**Pull request by hidalgoss**
Dirty working cpuinfo test. Please let your commets about.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    for(auto & item : vf_v1)
+        item = get_random_float();
+    for(auto & item : vf_v2)
+        item = get_random_float();
+    Suma mySum;
+    vd_sum = mySum.Sum_Two_Vector(vf_v1, vf_v2);
+    
+
+    named_log_debug("test debug message with a module name");
```
Comment (pablogh-2000) [inline src/main.cpp:49]: indentation!

File: src/main.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	nameless_log_debug("test debug message without a module name");
+	suppressed_log_debug("test suppressed debug message");
```
Comment (pablogh-2000) [inline src/main.cpp:51]: Can you dump variables with CLOG? If so, I want to see it.
I also want to know if the log dumps to std output or to logfile.

File: CMakeLists.txt [hunk 3]
```diff
@@ -3,16 +3,48 @@
 #
 cmake_minimum_required (VERSION 3.16.3)
 
-project (HelloWorld)
+set(PROJ_NAME_MAIN HelloWorld)
```
Comment (pablogh-2000) [inline CMakeLists.txt:6]: create a README with the required compilation instructions and do it comprehensive, meaning that ANYONE can install this repo.

File: src/main.cpp (hunk 1 above)
Comment (hidalgoss) [reply src/main.cpp]: Done? Please, Can you orient me? For helping reading sometimes let a blank line between code snippets to clarify meanings. Does it not a good practice for you?

File: CMakeLists.txt (hunk 3 above)
Comment (hidalgoss) [reply CMakeLists.txt:6]: Done. But directory design for dependencies, sources and headers are not the best approach so CMAKE still have incorrectness in this dummy project. I have solved making a new structure which will be used in next branch to be created to develop study of cpuinfo meanings exclusively.

File: test/Test_main.cpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include "helloworld.h"
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// Post: Redirect STDOUT and STDERR standard file descriptors to specified files.
+// Return: -1 if something fails or 0 if success.
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+int redirect_stdout_to_file(const char * pc_file_log, const char * pc_file_error){
```
Comment (pablogh-2000) [inline test/Test_main.cpp:16]: Remove this function

File: test/Test_main.cpp [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    close(i_fd_log);
+    close(i_fd_error);
+    return 0;
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// Post: Restore STDOUT and STDERR to default behaviour.
+// Return: -1 if something fails or 0 if success.
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+int redirect_stdout_to_file_restore(){
```
Comment (pablogh-2000) [inline test/Test_main.cpp:45]: Remove

File: test/Test_main.cpp [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    return 0;
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// Post: Robustness for ending unexècted rpogram ending.
+//       Restores STDOUT & STDERR to default bahaviour if crash due to any of below
+//       signals. 
+//       SIGTERM, SIGSEGV, SIGINT, SIGILL, SIGABRT, SIGFPE.
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+void signalHandler(int i_SigNum) {
```
Comment (pablogh-2000) [inline test/Test_main.cpp:60]: Remove

File: test/Test_main.cpp [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+    }
+}
+
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// https://google.github.io/googletest/quickstart-cmake.html
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+int main(int argc, char **argv) {
+    signal(SIGTERM, signalHandler);
```
Comment (pablogh-2000) [inline test/Test_main.cpp:86]: Remove all these signals

File: test/Test_main.cpp [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    signal(SIGILL, signalHandler);
+    signal(SIGABRT, signalHandler);
+    signal(SIGFPE, signalHandler);
+    
+    ::testing::InitGoogleTest(&argc, argv);
+    cpuinfo_initialize();
+#ifdef RESULTS_TO_FILE
+    const char * pc_file_log = "log.txt";
+    const char * pc_file_error = "error.txt";
+    redirect_stdout_to_file(pc_file_log, pc_file_error);
```
Comment (pablogh-2000) [inline test/Test_main.cpp:98]: Remove

**Review by pablogh-2000 — COMMENTED** [review]
Remove the JPG file

File: test/Test_main.cpp (hunk 4 above)
Comment (hidalgoss) [reply test/Test_main.cpp]: Done.

File: test/Test_main.cpp (hunk 5 above)
Comment (hidalgoss) [reply test/Test_main.cpp]: Done.

File: test/Test_main.cpp (hunk 6 above)
Comment (hidalgoss) [reply test/Test_main.cpp]: Done.

File: test/Test_main.cpp (hunk 7 above)
Comment (hidalgoss) [reply test/Test_main.cpp]: Done.

**Comment by hidalgoss** [comment]
Merge considerations followed.

helloworld.cpp
F(x) Sum_Two_Vector() with std::transform and exception thrown are in SUM branch thus in this case, we got Sum code and not cpuinfo, for this cases.

main.cpp 
Has been combined Sum & cpuinfo. cpuinfo had older std:.cout print version and Sum_Two_Vector() return management not by exception. So has been considered the Sum code to in this part to mantain this version.

Test_HelloWorld.cpp
Sum test solves requested requirements about eliminate Test Cases regarding sum of one unit and Bad Case which expects a wrong Sum adding 1+2. Rest of test cases has been mantained from Sum but still had not implemented Clog for prints.
Preserved too thrown exception management for Sum_Two_Vector() from Sum.
Added ALL cpuinfo Unit Test examples just after the Sum preserved ones.

File: include/helloworld.h [hunk 9]
```diff
@@ -1,21 +1,22 @@
-﻿#ifndef HELLOWORLD_H
```
Comment (pablogh-2000) [inline include/helloworld.h:1]: What happened here? both blocks show as different but I can see that they are identical. Please check the rebase

File: test/Test_HelloWorld.cpp [hunk 10]
```diff
@@ -123,4 +155,1528 @@ TEST(TestSuite_HelloWorld, Validate_Vectors_different_size_Test__GC) {
         std::cout << "CONTROLLED EXCEPTION: *" << e.what() << "*" << std::endl;
         EXPECT_TRUE(true);
     }
-}
\ No newline at end of file
+}
+
+
+TEST(PROCESSORS_COUNT, non_zero) {
```
Comment (pablogh-2000) [inline test/Test_HelloWorld.cpp:161]: Remove all CPUINFO tests. They don't add anything.

**Review by pablogh-2000 — APPROVED** [review]

**Pull request merged**